File: dapp/src/utils/constant.ts
import * as anchor from '@project-serum/anchor';

export const LOGIN_MESSAGE = 's3cretseedsforlogin';
export const API_BASE_URL = 'https://34.238.156.194.nip.io';
// export const API_BASE_URL = 'http://localhost:8000';
export const RPC_API_KEY =
    'https://sleek-blue-liquid.solana-devnet.quiknode.pro/36e54bb0b3be1f41d0d78439827c8bd97771533e/';

export const PROGRAM_ADDRESS = '7Ji9oG6aVk8SJutzwRP3nFZJYXtq7FaDqkVWvmBbc4z';

export const CAMPAIGN_SEED = 'campaign_seed';
export const CAMPAIGN_AUTHORITY_SEED = 'campaign_authority_seed';
export const DONOR_SEED = 'donor_seed';
export const PROPOSAL_SEED = 'proposal_seed';
export const VOTE_SEED = 'vote_seed';

export const USDC_MINT = new anchor.web3.PublicKey(
    '83DPRebGNu41Gr1ucibUtp538GuUJab8b1Quy5BkqDkf'
);

export const USDC_DECIMALS = 9;

export const STATUS_ACTIVE = 0;
export const STATUS_FILLED = 1;
export const STATUS_NOT_FILLED = 2;
export const STATUS_VOTING = 3;
export const STATUS_FUNDED = 4;
export const STATUS_NOT_FUNDED = 5;
export const STATUS_FUND_CLAIMABLE = 6;
export const STATUS_SUCCESS = 7;
export const STATUS_FAILED = 8;

export const STATUS_KYC_PENDING = 0;
export const STATUS_KYC_ACCEPTED = 1;
export const STATUS_KYC_DECLINED = 2;
export const STATUS_KYC_REMOVED = 3;

export const EVIDENCE_STATUS_WAITING = 0;
export const EVIDENCE_STATUS_REQUESTED = 1;
export const EVIDENCE_STATUS_SUCCESS = 2;
export const EVIDENCE_STATUS_FAILED = 3;

export const OPACITY = 0.5;

export const ITEM_PER_PAGE = 5;

File: contracts/programs/solcare/src/constant.rs
use anchor_lang::{constant, prelude::Pubkey};

#[constant]
pub const CAMPAIGN_SEED: &[u8] = b"campaign_seed";

#[constant]
pub const CAMPAIGN_AUTHORITY_SEED: &[u8] = b"campaign_authority_seed";

#[constant]
pub const DONOR_SEED: &[u8] = b"donor_seed";

#[constant]
pub const PROPOSAL_SEED: &[u8] = b"proposal_seed";

#[constant]
pub const VOTE_SEED: &[u8] = b"vote_seed";

#[constant]
pub const STATUS_ACTIVE: u8 = 0;

#[constant]
pub const STATUS_FILLED: u8 = 1;

#[constant]
pub const STATUS_NOT_FILLED: u8 = 2;

#[constant]
pub const STATUS_VOTING: u8 = 3;

#[constant]
pub const STATUS_FUNDED: u8 = 4;

#[constant]
pub const STATUS_NOT_FUNDED: u8 = 5;

#[constant]
pub const VOTING_DURATION_IN_DAYS: u8 = 3;

#[constant]
pub const USDC_MINT_PUBKEY: Pubkey = Pubkey::new_from_array([
    104, 146, 249, 200, 170, 10, 89, 118, 166, 205, 86, 177, 177, 71, 178, 18, 149, 129, 244, 116,
    13, 216, 69, 146, 239, 73, 202, 144, 19, 110, 8, 4,
]); // 83DPRebGNu41Gr1ucibUtp538GuUJab8b1Quy5BkqDkf

#[constant]
pub const USDC_DECIMALS: u8 = 9;
